generators/client/templates/angular/src/main/webapp/app/layouts/main/_main.component.ts

```diff
@@ -50,6 +50,15 @@ export class <%=jhiPrefixCapitalized%>MainComponent implements OnInit {
                     params = destinationEvent.params;
                     destinationData = destinationEvent.data;
                     destinationName = destinationEvent.url[0].path;
+                } else {
+                    destinationEvent = event.state.root.children[0];
```

Comment:
@sendilkumarn I'll try to check this during today or tomorrow, but yes for me we shouldn't need that, the guard should be sufficient.